StRoot/StFcsWaveformFitMaker/PeakAna.cxx

```diff
@@ -0,0 +1,1336 @@
+  if( mComputedIndex<static_cast<Int_t>(mPeaks.size()) ){
+    mFoundPeak = mPeaks.at(mComputedIndex); //No offset for now rethink in the future??
+    //@[May 10, 2022]>mFoundPeak is a copy of the one in the vector. When drawing the windows the one in the vector is accessed not 'mFoundPeak' which is why the color for the one in the vector is changed and not mFoundPeak.
+    //mPeaks.at(mComputedIndex).SetStartLineColor(kRed);
+    //mPeaks.at(mComputedIndex).SetEndLineColor(kOrange);
+  }
+  else{mFoundPeak.SetWindow(mXRangeMax+1,mXRangeMax+1);}//If no valid index then set found peak to greater than max X values so it is obvious something's wrong
+  delete noisyana;
+  return mComputedIndex;
+}
+
+bool PeakAna::ValidPeakIdx() const
+{
+  if( 0<=mComputedIndex && mComputedIndex<static_cast<Int_t>(mPeaks.size()) ){return true;}
+  else{return false;}
+}
+
+bool PeakAna::GoodWindow()
+{
+  if( mComputedIndex<0 ){this->AnalyzeForPeak();}
+  //First check if start and end times are within our max timebin window of 0-1023
+  if( mFoundPeak.mStartX<mXRangeMin || mFoundPeak.mStartX>mXRangeMax ){/*std::cout<<"StartOut"<<std::endl;*/return false;}
+  else if( mFoundPeak.mEndX<mXRangeMin || mFoundPeak.mEndX > mXRangeMax ){/*std::cout<<"EndOut"<<std::endl;*/return false;}
+  //Next check if values make physical sense
+  else if( mFoundPeak.mStartX==mFoundPeak.mEndX ){/*std::cout<<"Equal"<<std::endl;*/return false;}
+  else if( mFoundPeak.mStartX > mFoundPeak.mEndX ){/*std::cout<<"Start>End"<<std::endl;*/return false;}
+  else{return true;}
+}
+
+void PeakAna::GetXYMax(Double_t xmin, Double_t xmax)
+{
+  if( mG_Data==0 ){ return; }
+  for( int i=0; i<mG_Data->GetN(); ++i )
+    {
+      Double_t X; Double_t Y;
+      mG_Data->GetPoint(i,X,Y);
+      if( X<xmin || X>xmax ){continue;}
+      if( Y>mMaxY ){ mMaxY=Y; mMaxX=X; }
+    }
+  if( mMaxY<0 && mMaxX<0 )
+    {
+      std::cout << "Unable to find a maximum adc\nSetting to impossible values" << std::endl;
+      mMaxY = 5000;
+      mMaxX = 5000;
+    }
+}
+
+Double_t PeakAna::PeakX()
+{
+  if( mComputedIndex<0 ){ this->AnalyzeForPeak(); }
+  if( mFoundPeak.mP_Peak<0){ return 0; }
+  else{ Double_t x,y; mG_Data->GetPoint(mFoundPeak.mP_Peak,x,y); return x;}
+}
+
+Double_t PeakAna::PeakY()
+{
+  if( mComputedIndex<0 ){this->AnalyzeForPeak();}
+  if( mFoundPeak.mP_Peak<0){ return 0; }
+  else{ Double_t x,y; mG_Data->GetPoint(mFoundPeak.mP_Peak,x,y); return y;}
+}
+
+void PeakAna::PeakXY(Double_t &xval, Double_t &yval)
+{
+  if( mComputedIndex<0 ){this->AnalyzeForPeak();}
+  if( mFoundPeak.mP_Peak<0 ){ xval=0; yval=0; }
+  else{ mG_Data->GetPoint(mFoundPeak.mP_Peak,xval,yval); }
+}
+
+void PeakAna::Print(Option_t* opt) const
+{
+  std::cout << "|P::Graph:"<<mG_Data;
+  std::cout << "|RangeX:["<<mXRangeMin<<","<<mXRangeMax<<"]";
+  std::cout << "|RangeY:["<<mYRangeMin<<","<<mYRangeMax<<"]";
```

Comment:
See https://star-bnl.github.io/star-coding-guidelines/formatguide.xml#Printing_Messages_in_STAR

Reply:
I thought in a Print function such as this using `cout` should not be a problem since  calling a Print function means that you want it to print to the standard out. Are we not allowed to have `std::cout` anywhere and have to use the STAR handlers only?

Reply:
I am not aware of any exclusion to the rule requiring the use of LOG_ stream objects. The advantage is that we can turn off or limit logging to a predefined level. I hope in production only errors and maybe warnings are logged to avoid unnecessary writes to, for example, LOG_DEBUG and LOG_INFO streams

Reply:
I changed some cout statements so that it uses the STAR messenger system where appropriate. Let me know if there are other places where it needs to be changed.